### PR TITLE
adds kebab/param-cased version of file name as possible name for bundle file

### LINKED ISSUE #65264: Gradle build failed to produce an .aab file when multiple flavorDimensions specified
I was having trouble building a flutter app with `2 flavor dimensions` and `2 flavors` for each dimension (total of 4 variants, 
excluding the debug and release modes).

here's a minimal flutter app repo: https://github.com/brightknight08/flutter_build_appbundle_error to reproduce

when I do `flutter build appbundle --release --flavor signedprod`, it would display 
```clli

`flutter doctor -v`
```cli
PS E:\lab\flutter_build_appbundle_error> flutter doctor -v
[✓] Flutter (Channel stable, 1.20.2, on Microsoft Windows [Version 10.0.19041.450], locale en-US)
    • Flutter version 1.20.2 at C:\flutter
    • Framework revision bbfbf1770c (3 weeks ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at E:\asdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = E:\asdk
    • Java binary at: E:\as\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Android Studio (version 4.0)
    • Android Studio at E:\as
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code (version 1.48.2)
    • VS Code at C:\Users\tim_m\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.14.0

[✓] Connected device (1 available)
    • P[redacted]...droid-arm64 • Android 10 (API 29)

• No issues found!
PS E:\lab\flutter_build_appbundle_error>
```

### PR DESCRIPTION
**Pull request by PritamSangani**
This PR adds the kebab/param-cased version of the .aab bundle file name to the possible options so the gradle tool can look for it in the bundle directory and prevent the `flutter build` command failing because it could not find the .aab file and the end of the build.

*List which issues are fixed by this PR. You must list at least one issue.*
Fixes #65264

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 1]
```diff
@@ -35,36 +35,22 @@ import 'multidex.dart';
 /// The directory where the APK artifact is generated.
 Directory getApkDirectory(FlutterProject project) {
   return project.isModule
-    ? project.android.buildDirectory
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/android/gradle.dart:38]: Please revert all formatting changes here, per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting. It is difficult to review the diff when formatting changes are included, and we would not be merge the style changes anyway, as they violate our style guide. Thanks.

Comment (christopherfujino) [inline packages/flutter_tools/lib/src/android/gradle.dart:38]: Please revert all formatting changes here, per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting. It is difficult to review the diff when formatting changes are included, and we would not be merge the style changes anyway, as they violate our style guide. Thanks.

**Comment by PritamSangani** [comment]
@christopherfujino - sorry about that. I have pushed a commit to revert the formatting issues.

**Comment by christopherfujino** [comment]
Removing myself as a reviewer, this should get picked up the android platform team's PR triage.

**Pull request closed without merging**